### PR TITLE
#21879 orgunit addresses

### PR DESCRIPTION
**Pull request by danchr**
This implements the server-side logic for specifying addresses when creating of organisational units. While at it, I cleaned up the handling of phone numbers ever so slightly.

### CODE REVIEW

File: mora/service/address.py [hunk 1]
```diff
@@ -82,9 +106,21 @@ def get_address_class(c, addrrel, class_cache):
 
 def get_one_address(c, addrrel, class_cache=None):
     addrclass = get_address_class(c, addrrel, class_cache)
+    addrformat = addrclass['scope']
```
Comment (cmoesgaard) [inline mora/service/address.py]: ...And the same goes for this function as well, basically.

File: tests/test_integration_address.py [hunk 2]
```diff
@@ -675,24 +674,209 @@ def test_create_unit_address(self):
         # NOW CREATE IT
 
         self.assertRequestResponse(
-            '/service/ou/{}/create'.format(unitid),
+            '/service/ou/{}/edit'.format(unitid),
```
Comment (cmoesgaard) [inline tests/test_integration_address.py]: Is edit used to create addresses for org units?

File: mora/service/address.py [hunk 3]
```diff
@@ -55,9 +68,20 @@ def get_relation_for(classobj, value):
...
+
+        if scope == 'PHONE':
+            value = re.sub(r'\s+', '', value)
+
+            if not value.startswith('+'):
+                value = '+45' + value
+
+        if not util.is_urn(value):
+            value = prefix + value
+
```
[the remark this answers is not in the document]
Comment (danchr) [reply mora/service/address.py]: Well, the thing is that it'll be hard to find a format language that's sufficiently expressive; for example, we'll want to format some values as numbers, with zero padding, or prepend a country code to a phone number unless already specified.

Hence, I think certain hardcoded formats combined with a list of URN prefixes is better.

File: tests/test_integration_address.py (hunk 2 above)
Comment (danchr) [reply tests/test_integration_address.py]: From my understanding, `create` is used to create the unit itself, whereas all details are modified using the `edit` endpoint. Still, this test doesn't actually manage to add any addresses — considering that editing doesn't work yet — so I'll remove it.

Comment (cmoesgaard) [reply tests/test_integration_address.py]: As far as I can see, we use the `create` endpoint for creating addresses for employees, so I think we should try to along the workflow for `e` and `ou` as much as we can.

File: mora/service/address.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (danchr) [reply mora/service/address.py]: Alas, we have to have _two_ functions: one to append the prefix, and another one to remove it. Plus a third one for generating the URL. On the other hand, we really do have the case that most of our URNs are quite simple — apart from phone numbers.

I'd like to fix it, I just haven't had a good idea, yet.

[the remark this answers is not in the document]
Comment (cmoesgaard) [reply mora/service/address.py]: I'm at least thinking we'll reach a point where we have to implement specific address types for different customers.

I'm not necessarily saying it's something we should fix now - but I think we should at least start planning ahead for this do be done _at some point_.

**Pull request merged**